dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java

```diff
@@ -160,15 +162,76 @@ public void testImportStructure()
-            System.err.println(ex.getMessage());
-            System.exit(1);
+            StructBuilder.importStructure(context, input, outputDocument, false);
+        } finally {
+            context.restoreAuthSystemState();
+        }
+
+        // Compare import's output with its input.
+        // N.B. here we rely on StructBuilder to emit communities and
+        // collections in the same order as the input document.  If that changes,
+        // we will need a smarter NodeMatcher, probably based on <name> children.
+        Source output = new StreamSource(
+                new ByteArrayInputStream(outputDocument.toByteArray()));
+        Source reference = new StreamSource(
+                new ByteArrayInputStream(
+                        IMPORT_DOCUMENT.getBytes(StandardCharsets.UTF_8)));
+        Diff myDiff = DiffBuilder.compare(reference).withTest(output)
+                .normalizeWhitespace()
+//                .withNodeFilter(new MyNodeFilter())
+                .withAttributeFilter((Attr attr) ->
+                        !attr.getName().equals("identifier"))
+                .checkForIdentical()
+                .build();
+
+        // Was there a difference?
+        // Always output differences -- one is expected.
+        ComparisonFormatter formatter = new DefaultComparisonFormatter();
+        for (Difference difference : myDiff.getDifferences()) {
+            System.err.println(difference.toString(formatter));
+        }
+        // Test for *significant* differences.
+        assertFalse("Output does not match input.", isDifferent(myDiff));
+
+        // TODO spot-check some objects.
+    }
+
+    /**
+     * Test of importStructure method, with given Handles.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testImportStructureWithHandles()
+            throws Exception {
+        System.out.println("importStructure");
+
+        // Run the method under test and collect its output.
+        ByteArrayOutputStream outputDocument
+                = new ByteArrayOutputStream(IMPORT_DOCUMENT.length() * 2 * 2);
+        byte[] inputBytes = IMPORT_DOCUMENT.getBytes(StandardCharsets.UTF_8);
+        context.turnOffAuthorisationSystem();
+        try (InputStream input = new ByteArrayInputStream(inputBytes);) {
+            StructBuilder.importStructure(context, input, outputDocument, true);
         } finally {
             context.restoreAuthSystemState();
         }
 
+        // Check a chosen object for the right Handle.
+        boolean found = false;
+        for (Community community : communityService.findAllTop(context)) {
+            for (Handle handle : community.getHandles()) {
+                if (handle.getHandle().equals(COMMUNITY_1_HANDLE)) {
+                    found = true;
+                    break;
+                }
+                if (found) {
+                    break;
+                }
+            }
+        }
```

Comment:
is there a reason to apply a different test strategy than the one used in `testImportStructure` via the `DiffBuilder`?
checking for just a single handle could hide lot of (future) issues (maybe the import skip all the community without a handle or handle are not set for collection, etc.)

Reply:
I've added a spot-check for a Collection handle and a Community with handle not specified.

Below the cited code, this test also applies the strategy from `testImportStructure`.